packages/babel-parser/src/parser/statement.js

```diff
@@ -86,8 +118,14 @@ export default class StatementParser extends ExpressionParser {
   }
 
   parseStatementContent(declaration: boolean, topLevel: ?boolean): N.Statement {
-    const starttype = this.state.type;
+    let starttype = this.state.type;
     const node = this.startNode();
+    let kind;
+
+    if (this.isLet()) {
```

Comment:
If `declaration` is false and the identifier after `let` is on a new line, we shouldn't enter this `if`:

e.g.
```js
// VariableDeclaration (correct)
while (0) let
[]
```
```js
// ExpressionStatement + BlockStatement (correct)
while (0) let
{}
```
```js
// ExpressionStatement + ExpressionStatement (currently parsed as VariableDeclaration)
while (0) let
foo
```